### ISSUE TITLE
warning: corrupt incremental compilation artifact found at

### ISSUE THREAD

**Issue by Razzwan** [rust-lang/rust#148130]
I tried this code:

```rust
regular application
```

I expected to see this happen: no warning

Instead, this happened: 2 similar warning occures

### Meta
warning:
> warning: corrupt incremental compilation artifact found at `/app/api/target/debug/incremental/game_api-0pftixk2hoy7p/s-hce0fdbqzv-0us8nb3-working/dep-graph.bin`. This file will automatically be ignored and deleted. If you see this message repeatedly or can provoke it without manually manipulating the compiler's artifacts, please file an issue. The incremental compilation system relies on hardlinks and filesystem locks behaving correctly, and may not deal well with OS crashes, so whatever information you can provide about your filesystem or other state may be very relevant.

`rustc --version --verbose`:
```
rustc 1.90.0 (1159e78c4 2025-09-14)
binary: rustc
commit-hash: 1159e78c4747b02ef996e55082b704c09b970588
commit-date: 2025-09-14
host: x86_64-unknown-linux-gnu
release: 1.90.0
LLVM version: 20.1.8
```

**Comment by bjorn3** [comment]
Has this happened only once to you or have you seen this at multiple times different times. Have you had a system crash or hard restart recently? On what filesystem are the build artifacts stored? Ext4, btrfs, nfs, ...?

**Comment by Razzwan** [comment]
This problem happened many times, when I try to compile my code inside docker container `rust:1.90.0-slim-bullseye` with such script: `docker exec -e RUSTFLAGS="-C target-cpu=skylake -C target-feature=+aes,+pclmul,+sse4.2,+avx2" $(SERVICE_N)_api cargo build --release --target x86_64-unknown-linux-gnu`

> Have you had a system crash or hard restart recently?

If the system crash (which happens time by time), I can't reliably associate it with an error because the error is not visible. Terminal closed, and docker exited... No errors visible after that (Or I don't know, how to find them)

> On what filesystem are the build artifacts stored?

Ext4 (linux based on ubuntu)

**Comment by saethlin** [comment]
If the corrupt files are being caused by a crash, you would see this warning on the next compilation after turning the system back on.

... How often is your OS crashing?

**Comment by bjorn3** [comment]
Those generations have an issue where the cpu physically degrades much quicker than it should due to microcore requesting a too high cpu voltage. There are bios updates that should reduce the rate of degradation, but once it has degraded the only option to fix it is getting the cpu itself replaced. https://community.intel.com/t5/Mobile-and-Desktop-Processors/Intel-Core-13th-and-14th-Gen-Desktop-Instability-Root-Cause/m-p/1633442
